tests/functional/spec/list_collector_variants_primary_person.spec.js

```diff
@@ -0,0 +1,121 @@
+import VariantBlockPage from "../generated_pages/list_collector_variants_primary_person/variant-block.page";
+import PrimaryPersonListCollectorPage from "../generated_pages/list_collector_variants_primary_person/primary-person-list-collector.page";
+import ListCollectorAddPage from "../generated_pages/list_collector_variants_primary_person/list-collector-add.page";
+import ListCollectorPage from "../generated_pages/list_collector_variants_primary_person/list-collector.page";
+import EditPersonPage from "../generated_pages/list_collector_variants_primary_person/list-collector-edit.page";
+import SubmitPage from "../generated_pages/list_collector_variants_primary_person/submit.page";
+import ThankYouPage from "../base_pages/thank-you.page.js";
+
+describe("List collector with variants primary person", () => {
+  describe("Given that person is lives in house", () => {
+    before("Load the survey", () => {
+      browser.openQuestionnaire("test_list_collector_variants_primary_person.json");
+      it("The user is asked questions about whether they live there", () => {
+        $(VariantBlockPage.yes()).click();
+        $(VariantBlockPage.submit()).click();
+        expect($(PrimaryPersonListCollectorPage.legend()).getText()).to.contain("Do you live here? (variant)");
+      });
+    });
+  });
+  describe("Given the user starts on the 'Do you like variant' question", () => {
+    before("Load the survey", () => {
+      browser.openQuestionnaire("test_list_collector_variants_primary_person.json");
+    });
+    it("When the user says that they do live there, then they are shown as the primary person", () => {
+      $(VariantBlockPage.yes()).click();
+      $(VariantBlockPage.submit()).click();
+      $(PrimaryPersonListCollectorPage.youLiveHereYes()).click();
+      $(PrimaryPersonListCollectorPage.submit()).click();
+      $(ListCollectorAddPage.firstName()).setValue("John");
+      $(ListCollectorAddPage.lastName()).setValue("Doe");
+      $(ListCollectorAddPage.submit()).click();
+      expect($(ListCollectorPage.listLabel(1)).getText()).to.equal("John Doe (You)");
+    });
+    it("When the user adds another person, they are shown in the summary", () => {
+      $(ListCollectorPage.yesLabel()).click();
+      $(ListCollectorPage.submit()).click();
+      $(ListCollectorAddPage.firstName()).setValue("Samuel");
+      $(ListCollectorAddPage.lastName()).setValue("Clemens");
+      $(ListCollectorAddPage.submit()).click();
+      expect($(ListCollectorPage.listLabel(2)).getText()).to.equal("Samuel Clemens");
+    });
+    it("When the user goes back and answers No, the primary person is not shown", () => {
```

Comment:
I think we need to clarify what they are saying "No" i.e the respondent is changing their answer to No they do not live here